src/metaswitch/clearwater/etcd_shared/plugin_utils.py

```diff
@@ -25,12 +25,11 @@ def run_command(command, namespace=None, log_error=True):
     passing to this function.
     """
     if namespace:
-        command = "ip netns exec {} ".format(namespace) + command
+        command = "ip netns exec {} ".format(namespace) + shlex.quote(command)
 
     # Pass the close_fds argument to avoid the pidfile lock being held by
     # child processes
-    p = subprocess.Popen(command,
-                         shell=True,
+    p = subprocess.Popen(shlex.split(command),
```

Comment:
Does it need to be ```ip netns exec siganling echo hello```, as in the command ```echo hello``` needs to split up by shlex before appending them to the ["ip", "netns",...]?